server/src/main/java/org/spine3/server/entity/storage/ColumnType.java

```diff
@@ -0,0 +1,93 @@
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity.storage;
+
+import org.spine3.annotations.SPI;
+
+/**
+ * An interface for handling type conversion for the {@link Column Columns}.
+ *
+ * <p>When retrieved from an instance of an {@link org.spine3.server.entity.Entity Entity},
+ * the {@link Column Column} may be of an inappropriate type for storing. To convert the value into
+ * an appropriate type, use {@link #convertColumnValue}.
+ *
+ * <p>After the conversion, you might want to save the value into a special storage specific DTO,
+ * which commonly is built in a key-value fashion. To do that, call {@link #setColumnValue} and pass
+ * both the DTO to store the value to and the key to store it under.
+ *
+ * <p>Example of implementing this interface for a JDBC-based storage:
+ * <code>
+ *     <pre>
+ *         class VarcharDateType implements ColumnType<Date, String, PreparedStatement, Integer> {
+ *             \@Override
+ *             public String convert(Date fieldValue) {
+ *                 return MY_DATE_FORMAT.format(fieldValue);
+ *             }
+ *
+ *             \@Override
+ *             public void set(PreparedStatement storageRecord, String value, Integer columnIdentifier) {
+ *                 storageRecord.setString(columnIdentifier, value);
+ *             }
+ *         }
+ *     </pre>
+ * </code>
+ *
+ * <p>The example above translates a {@linkplain java.util.Date} into a formatted {@code String},
+ * which is persisted into the DB.
+ *
+ * <p>It's necessary to make these operations atomic to allow automatic type conversion when
+ * performing the DB queries.
+ *
+ * @param <J> the Java type represented by the column
+ * @param <S> the "store as" type of the column
+ * @param <R> the type of the record in the database, which holds a single cortege of data and
+ *            is consumed by the database upon write
+ * @param <C> the type of the column identifier in the {@code R}
+ */
+@SPI
+public interface ColumnType<J, S, R, C> {
+
+    /**
+     * Converts the {@link Column Column} specified in
+     * the {@link org.spine3.server.entity.Entity Entity} declaration to the type in which the Field
+     * is stored.
+     *
+     * <p>Common example is converting
+     * {@link com.google.protobuf.Timestamp com.google.protobuf.Timestamp} into
+     * {@link java.util.Date java.util.Date}.
+     *
+     * @param fieldValue the {@link Column Column} of the initial type
+     * @return the {@link Column Column} of the "store as" type
+     */
+    S convertColumnValue(J fieldValue);
```

Comment:
IDEA says this method is not used.

I would expect at least some tests to call it.